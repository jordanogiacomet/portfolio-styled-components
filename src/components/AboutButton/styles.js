import styled, { css } from 'styled-components';

export const Container = styled.button`
  background: transparent;
  position: relative;
  border: none;
  color: #ADB7BE;
  margin-right: 12px;
  cursor: pointer;
  padding: 0.5rem 1rem;
  &.light{
    color: black;
    ${({ active }) =>
    active &&
    css`
      color: #1E90FF;
      &::after{
        width: 100%;
      }
    `}
  }
  &:hover {
    color: #1E90FF;
  }
  &::after {
    content: '';
    display: block;
    height: 2px;
    background: linear-gradient(to right, #1E90FF, #4169E1, #00008B);
    transition: width 0.3s ease-in-out;
    width: 0;
    position: absolute;
    left: 0;
    bottom: 0;
  }
  &:hover::after {
    width: 100%;
  }
  @media (min-width: 768px) {
    font-size: 2rem;
  }
  @media (max-width: 640px) {
    font-size: 1.6rem;
  }
  ${({ active }) =>
    active &&
    css`
      color: #1E90FF;
      &::after{
        width: 100%;
      }
    `}
`;
