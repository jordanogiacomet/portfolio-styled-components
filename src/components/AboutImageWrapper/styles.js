import styled from 'styled-components';

export const Container = styled.div`
  margin-top: 4px;
  order: 2;
  > img {
  display: block;
  max-width: 100%;
  height: auto;
  border-radius: 10px; /* Ajuste o raio da borda conforme necess√°rio */
  box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1), 0 6px 20px rgba(0, 0, 0, 0.1);
  }
  
  @media (min-width: 768px) {
  margin-top: 0;
  }
`;
