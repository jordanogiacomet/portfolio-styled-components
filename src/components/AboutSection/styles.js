import styled from 'styled-components';

export const Container = styled.section`
  color: white;
  padding: 50px 0px;
  display: flex;
  flex-direction: column;
  @media (min-width: 768px) {
    display: grid;
    grid-template-columns: 1fr 1fr;
    gap: 50px;
    align-items: center;
    margin-top: 80px;
  }
  @media (min-width: 1280px) {
    gap: 50px;

  };
`;
