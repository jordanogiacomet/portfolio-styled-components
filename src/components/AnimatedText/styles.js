import styled from 'styled-components';
import { TypeAnimation } from 'react-type-animation';

export const Container = styled(TypeAnimation)`
  display: inline-block;
  
  &.light{
    color: ${({ theme }) => theme.colors.primaryColor}
  }
  &.dark{
    ${({ theme }) => theme.colors.secondaryColor}
  }
`;