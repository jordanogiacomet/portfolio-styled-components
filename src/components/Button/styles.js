import styled, { css } from 'styled-components';

const getButtonStyles = ({ theme, type }) => {
  const buttonType = theme.button[type] || theme.button.default;
  return css`
    padding: ${buttonType.padding};
    border-radius: ${buttonType.borderRadius};
    color: ${buttonType.textColor};
    background: ${buttonType.background};
    font-weight: bold;
    cursor: pointer; 
    display: inline-flex;
    align-items: center;
    justify-content: center;
    min-width: 150px; /* Ajuste a largura mínima conforme necessário */
    transition: scale ease-in-out 0.2s;
    &:hover {
      transition: scale ease-in-out 0.2s;
      scale: calc(1.03);
    };

    @media (${theme.media.lteMedium}) {
      width: 100%; /* Largura 100% em telas menores */
    }
  `;
};

export const Container = styled.button`
  ${getButtonStyles};
`; 