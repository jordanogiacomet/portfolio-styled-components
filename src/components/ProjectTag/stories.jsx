import ProjectTag from '.';

export default {
  title: 'ProjectTag',
  component: ProjectTag,
  args: {
    name: 'Web',
    isSelected: false,
  },
  argTypes: {
    name: { control: 'text' },
    isSelected: { control: 'boolean' },
    onClick: { action: 'clicked' },
  },
};

export const Default = (args) => <ProjectTag {...args} />;