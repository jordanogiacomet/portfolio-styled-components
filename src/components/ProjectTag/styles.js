import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ color }) => css`
    border: 2px solid ${props => (props.isSelected ? `${color}` : '#ADB7BE')};
    color: ${props => (props.isSelected ? `${color}` : '#ADB7BE')};
    
    &.light{
    border: 2px solid ${props => (props.isSelected ? `${color}` : '#000000')};
    color: ${props => (props.isSelected ? `${color}` : '#000000')};
    font-weight: 500;
  }
  `}
  background: transparent;
  
  padding: 0.5rem 1rem;
  border-radius: 9999px;
  cursor: pointer;
  transition: all 0.3s ease-in-out;
  
  
  

  &:hover {
    ${({ color }) => css`
      border-color: ${color};
      color: ${color};
    `}

  }
`;
