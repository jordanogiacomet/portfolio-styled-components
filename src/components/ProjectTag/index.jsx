import React from 'react';
import P from 'prop-types';
import * as Styled from './styles';
import { useColorMode } from '../../contexts/ColorModeContext';
import { useColorCustomization } from '../../contexts/ColorCustomizationContext';

const ProjectTag = ({ name, onClick, isSelected }) => {
  const { colorMode } = useColorMode();
  const { gradientColors } = useColorCustomization();
  return (
    <Styled.Container color={gradientColors.via} onClick={() => onClick(name)} isSelected={isSelected} className={colorMode}>
      {name}
    </Styled.Container>
  );
};

ProjectTag.propTypes = {
  name: P.string.isRequired,
  onClick: P.func.isRequired,
  isSelected: P.bool.isRequired,
};

export default ProjectTag;
