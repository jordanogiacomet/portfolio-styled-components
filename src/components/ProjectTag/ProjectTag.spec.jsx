// ProjectTag.spec.jsx
import { renderTheme } from '../../styles/render-theme';
import ProjectTag from '.';

describe('<ProjectTag />', () => {
  it('should match snapshot', () => {
    const { container } = renderTheme(<ProjectTag name="Test" onClick={() => {}} isSelected={false} />);
    expect(container).toMatchSnapshot();
  });

  it('should render ProjectTag with correct props', () => {
    const { getByText } = renderTheme(<ProjectTag name="Test" onClick={() => {}} isSelected={false} />);
    expect(getByText('Test')).toBeInTheDocument();
  });
});