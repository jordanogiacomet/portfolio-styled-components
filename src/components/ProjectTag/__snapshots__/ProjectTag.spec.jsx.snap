// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProjectTag /> should match snapshot 1`] = `
.c0 {
  border: 2px solid #ADB7BE;
  color: #ADB7BE;
  background: transparent;
  padding: 0.5rem 1rem;
  border-radius: 9999px;
  cursor: pointer;
  transition: all 0.3s ease-in-out;
}

.c0.light {
  border: 2px solid #000000;
  color: #000000;
}

.c0:hover {
  border-color: #4169E1;
  color: #4169E1;
}

<div>
  <div
    class="c0 dark"
    color="#4169E1"
  >
    Test
  </div>
</div>
`;
